packages/runtime-core/src/componentProps.ts

```diff
@@ -67,7 +67,7 @@ type PropMethod<T, TConstructor = any> = T extends (...args: any) => any // if i
 type RequiredKeys<T> = {
   [K in keyof T]: T[K] extends
     | { required: true }
-    | { default: any }
+    | { default: string | number | boolean | object | symbol | bigint; }
```

Comment:
default should be based on the `type` provided  or a factory which returns the same `type`

eg:
```ts
props: {
  test: {
    type: String,
    default: 11 // this is invalid
  }
}
```

Reply:
Thank you for your reminding, I will continue to optimize now